File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/FrontEnd/src/app/Module/MRole/Role.Entity.ts
import { IEntity } from "../IEntity.Entity";
//import { ModuleRoleEntity } from "../MModuleRole/ModuleRole.Entity";
import { PermissionEntity } from "../MPermission/Permission.Entity";

export class RoleEntity extends IEntity {
  Id: number = null;
  Code: string = null;
  Name: string = null;
  Cx: number = null;
  //ModuleRoleEntities = Array <ModuleRoleEntity> =[];
  PermissionEntities: Array<PermissionEntity> = [];
    public constructor() {
        super();
    }

}

File: sourceC# - Visual Studio/Workspaces/ESHOP/EShop/Entities/RoleEntity.cs
using EShop.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EShop.Entities
{
    public class RoleEntity : BaseEntity
    {
        public Guid Id { get; set; }
        public String Code { get; set; }
        public String Name { get; set; }
        public List<ModuleRoleEntity> ModuleRoleEntities { get; set; }
        public List<PermissionEntity> PermissionEntities { get; set; }

        public RoleEntity():base() { }

        public RoleEntity(Role Role, params object[] args) :base(Role)
        {
		    foreach(object arg in args)
			{
                if (arg is ICollection<ModuleRole> ModuleRoles)
                    ModuleRoleEntities = ModuleRoles.Select(model => new ModuleRoleEntity(model, model.Module)).ToList();				
                if (arg is ICollection<Permission> Permissions)
                    PermissionEntities = Permissions.Select(model => new PermissionEntity(model, model.Employee)).ToList();				
			}
        }
    }

    public class RoleSearchEntity : FilterEntity
    {
        public Guid? Id { get; set; }
        public String Code { get; set; }
        public String Name { get; set; }
    }
}
